tap_klaviyo_custom/streams.py

```diff
@@ -1,46 +1,214 @@
+import tap_klaviyo_custom.tap
+
+import singer
+
+LOGGER = singer.get_logger()
 
-# TODO: Delete this is if not using json files for schema definition
 SCHEMAS_DIR = Path(__file__).parent / Path("./schemas")
-# TODO: - Override `UsersStream` and `GroupsStream` with your own stream definition.
-#       - Copy-paste as many times as needed to create multiple stream types.
 
 
-class UsersStream(klaviyo-customStream):
+class ListsStream(RESTStream):
     """Define custom stream."""
-    name = "users"
-    path = "/users"
-    primary_keys = ["id"]
+    name = "lists"
+    path = "lists"
+    primary_keys = ["list_id"]
     replication_key = None
-    # Optionally, you may also use `schema_filepath` in place of `schema`:
-    # schema_filepath = SCHEMAS_DIR / "users.json"
-    schema = th.PropertiesList(
-        th.Property("name", th.StringType),
-        th.Property("id", th.StringType),
-        th.Property("age", th.IntegerType),
-        th.Property("email", th.StringType),
-        th.Property("street", th.StringType),
-        th.Property("city", th.StringType),
-        th.Property("state", th.StringType),
-        th.Property("zip", th.StringType),
-    ).to_dict()
-
-
-class GroupsStream(klaviyo-customStream):
+    schema_filepath = SCHEMAS_DIR / "lists.json"
+
+    url_base = "https://a.klaviyo.com/api/v2/"
+
+    records_jsonpath = "$[*]"  # Or override `parse_response`.
+
+    def get_url_params(self, partition: Optional[dict]) -> Dict[str, Any]:
+        """Return a dictionary of values to be used in URL parameterization."""
+        params = {}
+        params.update({"api_key": self.config['api_key']})
+        return params
+
+    def prepare_request(
+        self, context: Optional[dict], next_page_token: Optional[Any]
+    ) -> requests.PreparedRequest:
+        """Prepare a request object.
+
+        If partitioning is supported, the `context` object will contain the partition
+        definitions. Pagination information can be parsed from `next_page_token` if
+        `next_page_token` is not None.
+        """
+        http_method = self.rest_method
+        url: str = self.get_url(context)
+        params: dict = self.get_url_params(context)
+        request_data = self.prepare_request_payload(context, next_page_token)
+        headers = self.http_headers
+
+        request = cast(
+            requests.PreparedRequest,
+            self.requests_session.prepare_request(
+                requests.Request(
+                    method=http_method,
+                    url=url,
+                    params=params,
+                    headers=headers,
+                    json=request_data,
+                )
+            ),
+        )
+        return request
+
+    def get_url(self, context: Optional[dict]) -> str:
+        """Return a URL, optionally targeted to a specific partition or context.
+
+        Developers override this method to perform dynamic URL generation.
+        """
+        url = self.url_base+self.path
```

Comment:
If you are making the base url string a global variable, then make sure you make the changes at its every occurrence, as it will no longer be self.url_base